openpype/hosts/maya/api/lib.py

```diff
@@ -1325,8 +1326,8 @@ def generate_ids(nodes, asset_id=None):
 
     if asset_id is None:
         # Get the asset ID from the database for the asset of current context
-        project_name = legacy_io.active_project()
-        asset_name = legacy_io.Session["AVALON_ASSET"]
+        project_name = get_current_project_name()
+        asset_name = get_current_asset_name()
         asset_doc = get_asset_by_name(project_name, asset_name, fields=["_id"])
         assert asset_doc, "No current asset found in Session"
         asset_id = asset_doc['_id']
```

Comment:
This could just be `get_current_project_asset`?

Reply:
The variables `project_name` and `asset_name` are still used. There is not much advantage in changing it to
```
project_name = get_current_project_name()
asset_doc = get_current_project_asset()
asset_name = asset_doc["name"]
```
It is maybe more confusing...